### ISSUE TITLE
Validation in Lazy Mode refresh errors.

### ISSUE THREAD

**Issue by rennypoz** [livewire/livewire#95]
Attached to the issue the gif relative to the problem.

The code is the following:

Component
```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class FormWithValidation extends Component
{
    public $email;

    public function saveContact()
    {
        $validatedData = $this->validate([
            'email' => 'required|email|max:20',
        ]);

        $this->message = 'Saved!';
    }

    public function render()
    {
        return view('livewire.form-with-validation');
    }
}
```

View
```html
<div>
    Email: <input wire:model.lazy="email">

    @if($errors->has('email'))
        <span>{{ $errors->first('email') }}</span>
    @endif

    <button wire:click="saveContact">Save Contact</button>

    {{ $message }}
</div>
```

The Problem:

![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/9286706/62228470-0032d400-b3be-11e9-8cca-15b82a4a5909.gif)

I think there is a problem with the syncInput, after the callMethod the syncInput function has been called, and the errors are cleared.

**Comment by calebporzio** [comment]
Ok, I understand the problem. Here's what's happening:

1. Livewire is listening for the "change" event on the input (because of the ".lazy")
2. Livewire is listening for the "click" event on the button
3. When the input is still highlighted, and you click the button, here's what happens
  1. When the mouse presses down, the "change" event is fired and submits the "syncInput" to the server
  2. During the AJAX request, Livewire "locks" incoming actions
  3. When the mouse comes up, the "click" event is fired and wants to submit "callMethod", but there is a locked AJAX request out there so it never happens

Here are some potential solutions:
A) make `wire:click` an alias for `mousedown`
B) make ".lazy" be intelligent (like it was when I first built it). Instead of just listening for a change event, Livewire would queue the data modifications and send them along with the next request to the server

I think B is out because it would change the "lazy" behavior drastically and diverge from Vue's implementation.

Do you see any problems with A?

**Comment by rennypoz** [comment]
Hmmm but with mousedown isn't the same problem?

If the change event is fired on mousedown and livewire locks incoming actions, the next is the callMethod called from the same mousedown but it's still stuck because of the locked ajax.

Is my reasoning correct?

Ps, sorry for the poor english :)

**Comment by calebporzio** [comment]
Your reasoning is correct, however, Livewire actually queues actions that are fired within 5 ms. In that case, Livewire would receive both actions and fire them both off together.

The NEW problem is that now, the "callMethod" would get fired before the "syncInput".

I may end up just sorting actions by "syncInput" first... not sure though. Feels kinda icky.

**Comment by wsamoht** [comment]
Just ran into this myself today :)

**Comment by calebporzio** [comment]
If you change `wire:click` to `wire:mousedown`, the problem is now fixed in v0.3.15

**Comment by calebporzio** [comment]
I'm open to a solution that accounts for people using wire:click still, just can't think of anything simple off the top of my head.
